src/components/Search/index.tsx

```diff
@@ -0,0 +1,161 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+import './style.css';
+import { useState } from 'react';
+
+//Hooks
+import { useGet } from '../../hooks/get';
+
+//Components
+import Counter from '../../components/Counter';
+
+//DatePicker
+import DatePicker from 'react-datepicker';
+import "react-datepicker/dist/react-datepicker.css";
+
+
+//Imag
+import departure from '../../assets/Flights/departure.png'
+import arrival from '../../assets/Flights/arrival.png'
+import personsolid from '../../assets/Flights/personsolid.png'
+import calendarwithdates from '../../assets/Flights/calendarwithdates.png'
+import Button from '../../components/Button';
+import { useSearchContext } from '../../contexts/searchContext';
+
+export default function Search() {
+
+
+ /*********************Interface****************************/
+
+  interface typeOfStateClick{
+    [arg:string]:boolean,
+  }
+
+  /*********************State****************************/
+  
+  const [stateClick, setStateClick] = useState<typeOfStateClick>({
+    'fromWhere':false,
+    'whereTo':false,
+    'date':false,
+    'passenger':false,
+  });
+  const [checkbox, setCheckbox] = useState<'roundTrip' | 'oneWay'>('roundTrip');
+  const {setSearch} = useSearchContext();
+  const [fromWhere, setFromWhere] = useState<string>('From where?'); 
+  const [whereTo, setWhereTo] = useState<string>('Where To?'); 
+  const [adult, setAdult] = useState<number | "1 adult">("1 adult"); 
+  const [minors, setMinor] = useState<number>(0);   
+  const [startDate, setStartDate] = useState(new Date()); 
+  const [endDate, setEndDate] = useState(new Date()); 
+  const [defaultDate] = useState(new Date()); 
+
+ /*********************Method****************************/
+
+  const handleClick =(arg:string)=>{
+    setStateClick({[arg]:!stateClick[arg]}); 
+  }
+  const counterValueAdult =(counterValue:any)=>{
+    setAdult(counterValue); 
+  }  
+  const counterValueMinor =(counterValue:any)=>{
+    setMinor(counterValue); 
+  }
+  const handleSubmit =()=>{
+   setSearch([fromWhere , whereTo , adult , minors , startDate , endDate])
+  }
+
+
+ /*********************Api****************************/
+
+  const {data , isLoading , isSuccessful , message} = useGet(import.meta.env.VITE_API_FLIGHTNAMES);//fetch flight names
```

Comment:
As we said on the `.env` file, this is not the how nor the were you would import and use routes.